testutils/m/match.go

```diff
@@ -139,6 +151,20 @@ func MatchRoomTimelineMostRecent(n int, events []json.RawMessage) RoomMatcher {
 	}
 }
 
+func MatchRoomTimelineEndsWithID(wantID string) RoomMatcher {
```

Comment:
Please don't add new matchers which don't add new functionality, as it pollutes the number of functions you can use for assertions. To do this, you can just do (in tests-e2e/main_test.go): `MatchRoomTimelineMostRecent(1, []Event{{ID: "foo"}})`

Reply:
I can do. Though I'm pretty sure I've been down this road before; I naively expect that `MatchRoomTimelineMostRecent` wants me to provide a fully-specced out Event struct

Reply:
In E2E tests it's partial matches, as you don't know all the fields.

Reply:
e9ebe8a